client/engine/chainservice/mock_chainservice.go

```diff
@@ -58,6 +58,8 @@ func (mc *MockChainService) GetChainId() (*big.Int, error) {
 }
 
 func (mc *MockChainService) Close() error {
-	close(mc.txListener)
-	return mc.chain.Close()
+	if mc.txListener != nil {
+		close(mc.txListener)
+	}
```

Comment:
This jumps out a bit -- why is it necessary?

Reply:
Removed in e917c11c